### PR TITLE
refactor(capture): the file types, the sanitizer and the bounds are published once

### PR DESCRIPTION
**Pull request by tinnhlt-gra**
Phase 1a of the channel-attachments arc: give message channels the same attachment support email has, by building the seam **once** in the core so Telegram and each extension unit can adopt it without inventing their own.

This PR is the **refactor**, plus exactly one deliberate behaviour change, named below. No contract moves, no migration. It exists so the three PRs after it have one file type, one sanitizer and one set of bounds to build on instead of three.

## What moves, and why there

Three file types leave `internal/shared/ports/connector` for the published surface `backend/pkg/extension`, and are **aliased back** — the arrangement `ports/jurisdiction` already uses for the pack contract:

| published | aliased in core as |
|---|---|
| `extension.InboundFile` | `connector.Part` |
| `extension.FileDrop` | `connector.PartDrop` |
| `extension.OutboundFile` | `connector.OutboundFile` |

A file handed over by an extension unit and one handed over by a core connector must be literally the same Go type. The alternative — a mirrored type each side, kept honest by a parity test — is a second spelling of a file, which is a second set of bounds that can disagree about how large a file may be. `alias_test.go` compares `reflect.TypeOf` on each pair so a future edit cannot quietly replace an alias with a copy.

Alongside them, three helpers and four bounds move out of `capture/mailmap` and become published:

- `SniffContentType`, `DeclaredTypeDisagreement`, `SafeFilename` — because every producer of an `InboundFile` must answer "what is this file, and what is it called" the same way. A unit that self-reported `ContentType` would be certifying the one field whose entire purpose is to distrust the sender.
- `MaxInboundFiles` / `MaxInboundFilesExamined` / `MaxInboundFileBytes` / `MaxInboundMessageBytes`, aliased back into `mailmap` so there is one number per bound. **The aggregate is not optional**: publishing only the per-file bounds would license 20 × 25 MiB per message — ten times what mail admits — on code that reads as bounded, and it is what makes an in-memory `Body` safe. A test pins the four values and the one relationship that carries that weight — `MaxInboundMessageBytes < MaxInboundFiles × MaxInboundFileBytes` — because an aggregate raised to or past that product still reads as a bound while constraining nothing.

Finally, `extingressdrift_test.go`'s `Parts` waiver said the field is absent because *"there is no published path"* to the file keeper. As of this branch that is false, and the file's own rule is that a stale waiver "certifies nothing while reading as a decision". Rewritten to what it actually is now: a deliberate hold until the first unit that fills it (#1195), because freezing a published field no unit has exercised is the worst moment to freeze it.

## The evidence it is a refactor

The design predicted the alias would be transparent and named a stop condition: if `mailmap`, `gmail`, `sinkparts` or `comms` needed **any** edit after the move, a method or unkeyed literal exists that the analysis missed. It never fired — all four compiled and passed untouched. Every existing test in the tree passes unchanged.

## Two deviations from the plan, both to protect that claim

1. **The plan's `truncateRunes` was rejected.** It counts runes; the code it replaces (`mailmap.truncate`) counts **bytes** with a rune-boundary backoff, and `SafeFilename`'s guard is on bytes too. A 100-rune / 250-byte filename is cut today and would not have been under the plan's version — a silent behaviour change in a PR whose whole claim is that nothing changes. Moved verbatim as `truncateFilename` instead; `mailmap.truncate` stays because `mailmap.go:292` still calls it.

2. **The plan's `SniffContentType(nil)` expectation was wrong.** It asserts `application/octet-stream`; `http.DetectContentType(nil)` actually returns `text/plain`, and has done so through `mailmap.sniff` all along. The test records the real behaviour and says why it is surprising, plus a second case pinning what genuinely *does* sniff as octet-stream. Not "fixed" — an empty body typed as text is pre-existing and out of a refactor's scope.

One thing the plan did not anticipate: `mailmap/parts_test.go` held three tests over the moved helpers, which stopped compiling. They **moved with the logic** rather than reaching across a package boundary, and `pkg/extension/files_test.go` asserts every case they did (including `/////` and the header-injection name) plus the sniff window, parameter/case handling in the disagreement, and UTF-8 validity of a multi-byte cut.

## The one behaviour change

`SafeFilename` never stripped **U+2028 LINE SEPARATOR** or **U+2029 PARAGRAPH SEPARATOR**. `unicode.IsControl` answers for the `Cc` block only — CR, LF and NEL are `Cc` and were removed; U+2028/2029 are categories `Zl` and `Zp`, and `IsControl` returns `false` for both. The doc comment's claim that control characters stop a name from rewriting a log line was therefore **false**: `note<U+2028>X-Injected: yes.pdf` landed in the column intact and breaks the line in a log record, a JSON string or a CSV export exactly as a bare newline would.

Pre-existing on `main` — `mailmap.SafeFilename` has always had it — so the move did not introduce it. Fixed here because this PR publishes the function every future extension unit will call, and shipping it publishes the gap. The regression test was proved to fail with the `Zl`/`Zp` check removed before it was accepted, and the comment now says why that second check is not the redundancy it looks like.

Not a security advisory: no cross-tenant read, no row-scope or RBAC escape, no injection into a query — log and display confusion, i.e. hardening, fixed in the same change that found it.

## Review round 1

Fable (craft) and Codex (adversarial) both reviewed the branch; see [the review comment](#issuecomment-5341343283) for the full account. Codex could not refute the refactor claim across six equivalence traces. Fable found six things, all real, all fixed in `13366da` — the sharpest being that the bounds parity test **could not fail**: it compared an alias to its own definition, which compiles to `20 != 20`. The plan prescribed that test as written and the plan was wrong; the assertions now pin values and the load-bearing inequality instead.

## Gates

- `make check` — green (exit 0).
- `craft static --strict` on all 10 changed Go files — **0 blocker, 0 major, 0 minor**.
- New-code coverage on `pkg/extension/files.go` — **89.9%**. The two uncovered branches are the unreachable `return full` fallback and `truncateFilename`'s early return, both defensive guards.
- One staticcheck finding fixed en route: ST1018 refuses a literal U+202E in a string, so the RTL-override test case writes `‮`.

## Not applicable here

- **No UAT recording.** There is no user-visible surface to record: no endpoint, no screen, no MCP tool, no wire format changed. The one behaviour change is a filename sanitizer edge case, and the evidence for it is a regression test demonstrated to fail without the fix — which is stronger than a video, since no recording could show a U+2028 that is invisible by construction.
- **aicert not re-run.** The diff touches no prompt, no AI task and no routing config, so the pin cannot have moved.

## What follows

PR 2 widens the attachment category vocabulary (channel files currently could only land recorded as `email_attachment`, a hardcoded SQL literal under a DB CHECK). PR 3 turns `AttachmentCarrier` into a `Carriage()` descriptor and publishes it on `/v1/channel-providers`. Design and plan: `.tmp/channel-attachments/`.

## Summary by CodeRabbit

* **New Features**
  * Added shared attachment handling for inbound and outbound files.
  * Added content-type detection and warnings when declared types differ from detected types.
  * Added safer filename processing, including removal of unsafe characters and UTF-8-safe truncation.
  * Standardized inbound attachment and message size limits.

* **Documentation**
  * Updated extensibility documentation to describe shared attachment types, limits, and processing behavior.

* **Tests**
  * Added coverage for attachment validation, filename safety, type detection, and size limits.

### CODE REVIEW

**Comment by tinnhlt-gra** [comment]
## Review round 1 — Fable (craft) + Codex (adversarial), six findings, all fixed

**Codex could not refute the "pure refactor, no observable behaviour change" claim.** It traced `SafeFilename`, `SniffContentType` and `DeclaredTypeDisagreement` against their pre-move originals line by line, checked that the three deleted `mailmap` tests lost no assertion, confirmed the aliases are true Go aliases with no method sets or unkeyed literals, and found no layering or licence problem. Caveat it raised itself: it stayed read-only and ran no tests, so that is a static trace, not a substitute for the suite — which is green.

Both reviewers independently endorsed rejecting the plan's rune-based `truncateRunes`.

Fable found six things. **Every one checked out**, and `13366da` fixes them all.

### 1. The bounds parity test could not fail — this is the one that mattered

`maxParts` is *declared as* `extension.MaxInboundFiles`, so `if maxParts != extension.MaxInboundFiles` compiled to `if 20 != 20`. Assertion-free (T11): it could only ever fail by panicking, while its comment promised a guard the code did not contain. **The plan prescribed this test as written; the plan was wrong.**

`bounds_test.go` is deleted and the assertions moved next to the constants, where they pin the four **values** rather than a symbol against itself — plus the one relationship between the four numbers that is actually load-bearing and was untested anywhere:

```go
MaxInboundMessageBytes < MaxInboundFiles * MaxInboundFileBytes
```

Raise the aggregate to or past that product and it still exists, still reads as a bound, and constrains nothing — one message could hold 500 MiB of bodies at once. That is the assertion that makes `InboundFile.Body []byte` safe, and any single-constant edit can silently break it.

The general lesson, which would slip through anywhere else in this tree: **a parity test between an alias and its own definition is always green.** The compiler already makes that guarantee. What needs asserting is the value, and the relationships between values.

### 2. `(#1195)` pointed at a different problem

#1195 is *"A unit's poll has no way to report a record it could not represent"* — a whole **record** the core's grammar refuses, and the missing `Disposition` for it. The held `Parts`/`PartDrops` fields are about **attachments** refused by size bounds. A reader following the ref would have found a different problem and concluded the waiver was stale again — worse than no ref at all. Dropped; the reason already names its own trigger ("lands in the PR of the first unit that fills it") and stands alone. No new issue filed: that gap is planned work this arc closes, not an unfiled finding.

### 3. The sink comment described a lane that does not exist

"a channel fetch lane bounds before it downloads" is Phase 1c code, not in this tree — speculative narration (T3/T8) in the one PR that claims to change nothing. Rewritten to the rule that holds today: bounds belong to whichever producer reads the bytes, because bounding is what lets it read them at all.

### 4. Two comments argued opposite positions

`files.go` says a second spelling of a file is a second set of bounds that can disagree; `mailmap.recordParts` said the two shapes are *"deliberately separate types… a shared struct would let a future parser widen the seam"*. Both defensible, but together they left the next editor no way to tell which rule binds when adding a field. Reconciled: the line between `mailmap.Part` and `connector.Part` is a **stage**, not an owner — a candidate mid-walk that the collector can still refuse, versus a file that cleared every bound — which is categorically not the two-producers-two-bounds case the published rule forbids.

### 5. `DOC-PARAM-8` was silently dropped

It rode on `mailmap.SafeFilename`'s doc comment and its test; both were deleted and neither replacement picked it up, while DOC-PARAM-3/4/5/9 and DOC-AC-12/13 all survived the same move. Oversight, not decision. Restored on both.

### 6. `docs/explanation/extensibility.md` was left wrong about its own subsystem

Its published-surface enumeration reads as closed and omitted the file types, the four bounds and the sniff/sanitize pair — all importable by a unit as of this branch — and the "Where the code lives" table had the same hole. Both updated. Per CLAUDE.md a doc wrong about a subsystem is a defect in that subsystem, and this is the page a new extension author reads first.

---

`make check` exit 0 · `craft static --strict` 0 blocker / 0 major / 0 minor on all 10 changed files.

File: backend/pkg/extension/files.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+// "gpj.exe" with a RIGHT-TO-LEFT OVERRIDE before it renders as "...jpg".)
+func SafeFilename(name string, ordinal int) string {
+	cleaned := strings.Map(func(r rune) rune {
+		switch {
+		case r == '/' || r == '\\' || r == 0:
+			return -1
+		case unicode.IsControl(r):
+			return -1
+		case r >= 0x202A && r <= 0x202E, r >= 0x2066 && r <= 0x2069, r == 0x200F, r == 0x200E:
+			return -1
```
[the remark this answers is not in the document]
Comment (tinnhlt-gra) [reply backend/pkg/extension/files.go]: Confirmed and fixed in `8963c8e` — this one was real, and it is the finding I would have missed.

Verified directly rather than taken on trust: `unicode.IsControl` answers for the **Cc** block only. CR, LF and NEL are Cc and were stripped; U+2028 and U+2029 are **Zl** and **Zp**, and `IsControl` returns `false` for both. So the doc comment's claim that control characters stop a name from rewriting a log line was **false** — `note<U+2028>X-Injected: yes.pdf` landed in the column intact and breaks the line in a log record, a JSON string or a CSV export exactly as a bare newline would, which the same function already refuses.

This is **pre-existing on `main`** — `mailmap.SafeFilename` has always had it — so it is not something the move introduced. Fixed here anyway, which makes this PR a refactor plus one deliberate, named behaviour change: the function is already moving in this PR, and the published surface is what every future extension unit will call, so shipping the gapped version publishes the gap. Deferring would also have meant shipping a comment that says "control characters" while meaning "some control characters", which this repo's review-loop rule 5 forbids.

Two things beyond the literal fix:

- The doc comment now states that the line-break class is **two** tests and why the second is not redundant — `IsControl already covers control characters` is precisely the simplification that would reopen this.
- The regression test was **proved to fail** before it was accepted: with the Zl/Zp check removed it reports `SafeFilename("note\u2028X-Injected: yes.pdf") = "note\u2028X-Injected: yes.pdf"`. Added alongside it: CR and NEL (to pin what `IsControl` does cover) and an ordinary space (to pin what must **not** be stripped — U+2028/2029 are `unicode.IsSpace`, so a coarser space-based fix would have eaten real filenames).

Not routed to a security advisory: no cross-tenant read, no row-scope or RBAC escape, no injection into a query. It is log and display confusion — hardening rather than a live exploit — and it is fixed in the same change that found it.
